client/actions/SearchActions.js

```diff
@@ -0,0 +1,59 @@
+import axios from 'axios';
+import jwtDecode from 'jwt-decode';
+import { displayUserDocuments, displayDocuments } from './DocumentsActions';
+import { getUsers } from './UserActions';
+
+const displaySearchResults = (searchQuery, location, offset) => {
+  const userToken = localStorage.getItem('jwtToken');
+  const userData = jwtDecode(userToken);
+  const userId = userData.id;
+  return (dispatch) => {
+    dispatch({
+      type: 'IS_SEARCH',
+      searchPayload: {
+        isSearch: true,
+        searchQuery
+      }
+    });
+    if (location.match(/documents/)) {
+      axios.get(`/api/search/users/${userId}/documents/?q=${searchQuery}`)
+        .then((res) => {
+          if (searchQuery) {
+            dispatch({
+              type: 'DISPLAY_USER_DOCUMENTS',
+              documents: res.data
+            });
+          } else {
+            dispatch(displayUserDocuments());
+          }
+        });
+    } else if (location.match(/explore/)) {
+      axios.get(`/api/search/documents/?q=${searchQuery}&offset=${offset}`)
+        .then((res) => {
+          if (searchQuery) {
+            dispatch({
+              type: 'DISPLAY_DOCUMENTS',
+              documents: res.data
+            });
+          } else {
+            dispatch(displayDocuments());
+          }
+        });
+    } else if (location.match(/dashboard/)) {
```

Comment:
code from line 18 to this point seems to be repeated. The only difference is the action type. Why not abstract the code into a function to avoid repetition